### PR TITLE
feat: update `naics_classification` to latest version

### PR DESCRIPTION
**Pull request by jdhoffa**
This PR updates the `NAICS` dataset to the latest version (published in "2022"). 

Relates to #344 

@jacobvjk would appreciate a content review of this (especially on if some of these sectors should be Borderline). Steel in particular I was very unsure of.

### CODE REVIEW

**Comment by jdhoffa** [comment]
For posterity, the original raw data was pulled from here: https://www.census.gov/naics/?48967

**Comment by jdhoffa** [comment]
@jacobvjk updated the bridge to include all levels. 
This PR is ready for review now

File: data-raw/classification_bridge.R [hunk 1]
```diff
@@ -80,16 +80,49 @@ nace_classification <- dplyr::select(
...
   naics_classification_raw,
-  description = "naics_title",
+  sector = dplyr::case_when(
+    grepl("^3361", code) ~ "automotive",
+    grepl("^4811", code) ~ "aviation",
+    grepl("^481111", code) ~ "aviation",
+    grepl("^481211", code) ~ "aviation",
+    grepl("^3273", code) ~ "cement",
+    grepl("^21211", code) ~ "coal",
+    grepl("^2111", code) ~ "oil and gas",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:99]:
```suggestion
    grepl("^211", code) ~ "oil and gas",
    code %in% c("213111", "213112") ~ "oil and gas",
```

File: data-raw/classification_bridge.R [hunk 2]
```diff
@@ -80,16 +80,49 @@ nace_classification <- dplyr::select(
...
+naics_classification <- dplyr::mutate(
   naics_classification_raw,
-  description = "naics_title",
+  sector = dplyr::case_when(
+    grepl("^3361", code) ~ "automotive",
+    grepl("^4811", code) ~ "aviation",
+    grepl("^481111", code) ~ "aviation",
+    grepl("^481211", code) ~ "aviation",
+    grepl("^3273", code) ~ "cement",
+    grepl("^21211", code) ~ "coal",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:98]:
```suggestion
    grepl("^2121", code) ~ "coal",
    code == "213113" ~ "coal",
```

File: data-raw/classification_bridge.R [hunk 3]
```diff
@@ -80,16 +80,49 @@ nace_classification <- dplyr::select(
...
+    TRUE ~ "not in scope"
+  ),
+  borderline = dplyr::case_when(
+    code == "3361" ~ TRUE,
+    grepl("^33612", code) ~ TRUE,
+    code %in% c("4811", "481112") ~ TRUE,
+    grepl("^32732", code) ~ TRUE,
+    grepl("^32733", code) ~ TRUE,
+    grepl("^32739", code) ~ TRUE,
+    code == "213113" ~ TRUE,
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:112]:
```suggestion
   grepl("^21311", code) ~ TRUE,
```

File: data-raw/classification_bridge.R [hunk 4]
```diff
@@ -80,16 +80,49 @@ nace_classification <- dplyr::select(
...
-  description = "naics_title",
+  sector = dplyr::case_when(
+    grepl("^3361", code) ~ "automotive",
+    grepl("^4811", code) ~ "aviation",
+    grepl("^481111", code) ~ "aviation",
+    grepl("^481211", code) ~ "aviation",
+    grepl("^3273", code) ~ "cement",
+    grepl("^21211", code) ~ "coal",
+    grepl("^2111", code) ~ "oil and gas",
+    grepl("^22111", code) ~ "power",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:100]:
```suggestion
    grepl("^22111", code) ~ "power",
    code == "2211" ~ "power",
```

File: data-raw/classification_bridge.R [hunk 5]
```diff
@@ -80,16 +80,49 @@ nace_classification <- dplyr::select(
...
+  ),
+  borderline = dplyr::case_when(
+    code == "3361" ~ TRUE,
+    grepl("^33612", code) ~ TRUE,
+    code %in% c("4811", "481112") ~ TRUE,
+    grepl("^32732", code) ~ TRUE,
+    grepl("^32733", code) ~ TRUE,
+    grepl("^32739", code) ~ TRUE,
+    code == "213113" ~ TRUE,
+    TRUE ~ FALSE
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:113]:
```suggestion
    code == "2211" ~ TRUE,
    code %in% c("331512", "331513") ~ TRUE,
    TRUE ~ FALSE
```

File: data-raw/classification_bridge.R [hunk 6]
```diff
@@ -80,16 +80,49 @@ nace_classification <- dplyr::select(
...
+    grepl("^2111", code) ~ "oil and gas",
+    grepl("^22111", code) ~ "power",
+    grepl("^483", code) ~ "shipping",
+    code %in% c("331110", "331512", "331513") ~ "steel",
+    TRUE ~ "not in scope"
+  ),
+  borderline = dplyr::case_when(
+    code == "3361" ~ TRUE,
+    grepl("^33612", code) ~ TRUE,
+    code %in% c("4811", "481112") ~ TRUE,
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:108]:
```suggestion
    code %in% c("4811", "48111", "481112") ~ TRUE,
    code %in% c("4812", "48121", "481212", "481219") ~ TRUE,
```

**Review by jacobvjk — CHANGES_REQUESTED** [review]
A few changes I would suggest mostly relating to borderline inclusion.

In general, I am still unsure about how close to the core business we want a code to be in order to be considered `borderline == TRUE`. Especially in steel production there are a number of products that require high temperature furnaces in their processes. But I am unsure if this means that companies are often integrated

**Review by jacobvjk — COMMENTED** [review]
updated suggestions for steel based on https://www.transitionmonitor.com/wp-content/uploads/2020/09/steel_ef_final_pacta_banks_v1.2_14-09-2020.pdf and https://ieaghg.org/docs/General_Docs/Reports/PH3-30%20iron-steel.pdf

File: data-raw/classification_bridge.R [hunk 7]
```diff
@@ -80,16 +80,49 @@ nace_classification <- dplyr::select(
...
+    grepl("^3361", code) ~ "automotive",
+    grepl("^4811", code) ~ "aviation",
+    grepl("^481111", code) ~ "aviation",
+    grepl("^481211", code) ~ "aviation",
+    grepl("^3273", code) ~ "cement",
+    grepl("^21211", code) ~ "coal",
+    grepl("^2111", code) ~ "oil and gas",
+    grepl("^22111", code) ~ "power",
+    grepl("^483", code) ~ "shipping",
+    code %in% c("331110", "331512", "331513") ~ "steel",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R]:
```suggestion
    grepl("^3311", code) ~ "steel",
    grepl("^3312", code) ~ "steel",
    code %in% c("331512", "331513") ~ "steel",
```

File: data-raw/classification_bridge.R (hunk 5 above)
Comment (jacobvjk) [reply data-raw/classification_bridge.R:120]: this is still open

File: data-raw/classification_bridge.R [hunk 8]
```diff
@@ -80,16 +80,56 @@ nace_classification <- dplyr::select(
...
+    grepl("^4812", code) ~ "aviation",
+    grepl("^3273", code) ~ "cement",
+    grepl("^2121", code) ~ "coal",
+    code == "213113" ~ "coal",
+    grepl("^211", code) ~ "oil and gas",
+    code %in% c("213111", "213112") ~ "oil and gas",
+    grepl("^22111", code) ~ "power",
+    code == "2211" ~ "power",
+    grepl("^483", code) ~ "shipping",
+    grepl("^3311", code) ~ "steel",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:104]:
```suggestion
    grepl("^3312", code) ~ "steel",
```

**Review by jacobvjk — CHANGES_REQUESTED** [review]
somehow my latest steel suggestions were lost in the process

**Comment by jdhoffa** [comment]
> somehow my latest steel suggestions were lost in the process

ok this turned out very difficult to ensure i did correctly with half being "commit suggestions" and half just being "this is still open". 

Tried to manually update it as you suggested but i may have made a mistake.

**Comment by jacobvjk** [comment]
> > somehow my latest steel suggestions were lost in the process
> 
> ok this turned out very difficult to ensure i did correctly with half being "commit suggestions" and half just being "this is still open".
> 
> Tried to manually update it as you suggested but i may have made a mistake.

I will give it another full review just to be sure

File: data-raw/classification_bridge.R [hunk 9]
```diff
@@ -80,16 +80,57 @@ nace_classification <- dplyr::select(
...
+    grepl("^3312", code) ~ "steel",
+    code %in% c("331512", "331513") ~ "steel",
+    TRUE ~ "not in scope"
+  ),
+  borderline = dplyr::case_when(
+    code == "3361" ~ TRUE,
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:110]:
```suggestion
    code %in% c("336", "3361") ~ TRUE,
```

File: data-raw/classification_bridge.R [hunk 10]
```diff
@@ -80,16 +80,57 @@ nace_classification <- dplyr::select(
...
+    grepl("^33612", code) ~ TRUE,
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:111]:
```suggestion
    grepl("^33612", code) ~ TRUE,
    grepl("^3362", code) ~ TRUE,
    grepl("^3363", code) ~ TRUE,
```

**Review by jacobvjk — CHANGES_REQUESTED** [review]
some broader borderline definitions

File: data-raw/classification_bridge.R [hunk 11]
```diff
@@ -80,16 +80,57 @@ nace_classification <- dplyr::select(
...
+    grepl("^3361", code) ~ "automotive",
+    grepl("^4811", code) ~ "aviation",
+    grepl("^4812", code) ~ "aviation",
+    grepl("^3273", code) ~ "cement",
+    grepl("^2121", code) ~ "coal",
+    code == "213113" ~ "coal",
+    grepl("^211", code) ~ "oil and gas",
+    code %in% c("213111", "213112") ~ "oil and gas",
+    grepl("^22111", code) ~ "power",
+    code == "2211" ~ "power",
```
[the remark this answers is not in the document]
Comment (jdhoffa) [reply data-raw/classification_bridge.R:102]: What does this mean?

File: data-raw/classification_bridge.R [hunk 12]
```diff
@@ -80,16 +80,57 @@ nace_classification <- dplyr::select(
...
+  col_types = "_cc__",
+  skip = 2
 )
 
-naics_classification <- dplyr::select(
+naics_classification <- dplyr::mutate(
   naics_classification_raw,
-  description = "naics_title",
+  sector = dplyr::case_when(
+    grepl("^3361", code) ~ "automotive",
```
[the remark this answers is not in the document]
Comment (jdhoffa) [reply data-raw/classification_bridge.R]: This seems to not be correct:

Screenshot 2024-03-20 at 09 33 11

[the remark this answers is not in the document]
Comment (jdhoffa) [reply data-raw/classification_bridge.R]: Changed it to:
``` r
    grepl("^3361", code) ~ "automotive",
    grepl("^3362", code) ~ "automotive",
    grepl("^3363", code) ~ "automotive",
```

File: data-raw/classification_bridge.R (hunk 11 above)
[the remark this answers is not in the document]
Comment (jacobvjk) [reply data-raw/classification_bridge.R]: "see above". but you addressed this now

File: data-raw/classification_bridge.R (hunk 12 above)
[the remark this answers is not in the document]
Comment (jacobvjk) [reply data-raw/classification_bridge.R:93]: woops, definitely a bit too broad

File: data-raw/classification_bridge.R [hunk 13]
```diff
@@ -80,16 +80,63 @@ nace_classification <- dplyr::select(
...
+    code == "3273" ~ TRUE,
+    grepl("^32732", code) ~ TRUE,
+    grepl("^32733", code) ~ TRUE,
+    grepl("^32739", code) ~ TRUE,
+    code %in% c("213111", "213112") ~ TRUE,
+    code == "213113" ~ TRUE,
+    code == "2211" ~ TRUE,
+    grepl("^22112", code) ~ TRUE,
+    grepl("^3312", code) ~ TRUE,
+    code %in% c("331512", "331513") ~ TRUE,
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:126]:
```suggestion
```

**Review by jacobvjk — CHANGES_REQUESTED** [review]
Last change. We validated that casting steel in foundries is in-scope and therefore these two codes should not be borderline

**Review by jacobvjk — APPROVED** [review]
lgtm

**Pull request merged**